### PR TITLE
Add the ability to setRegexp function that allows custom function and with regular expression

### PR DESCRIPTION
**Pull request by ouchxp**
Add the ability to setRegexp function that allows custom function and with regular expression
Related issue:
https://github.com/arasatasaygin/is.js/issues/18
https://github.com/arasatasaygin/is.js/issues/15

### CODE REVIEW

**Comment by jdalton** [comment]
This PR changes so much it's hard to follow. For PRs try to keep them as simple and focused as possible.

**Comment by jdalton** [comment]
Feel free to open a similar more focused PR.

**Comment by ouchxp** [comment]
Sorry, I don't remember what I was thinking of when doing this PR. I'll create a new one shortly.
